src/main/java/frc/robot/subsystems/PowerMonitor.java

```diff
@@ -0,0 +1,155 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.hal.PowerDistributionFaults;
+import edu.wpi.first.wpilibj.PowerDistribution;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.Constants;
+import frc.robot.utils.logging.Logger;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class PowerMonitor extends SubsystemBase {
+    private final String loggingBase = "/robot/pdh/";
+    private final PowerDistribution pdh;
+    private final Set<Integer> monitoringChannels = new HashSet<>();
+    private final List<ChannelCurrentEvent> maxCurrentEvents = new ArrayList<>();
+    private final List<ChannelCurrentEvent> minCurrentEvents = new ArrayList<>();
+
+    public PowerMonitor() {
+        this.pdh = new PowerDistribution(1, PowerDistribution.ModuleType.kRev);
+    }
+
+    /**
+     * @param channel to monitor
+     * @return boolean representing if addition to monitoring set was successful
+     */
+    public boolean logChannel(int channel) {
+        if (channel < pdh.getNumChannels()) {
+            return monitoringChannels.add(channel);
+        }
+        return false;
+    }
+
+    /**
+     * @param channel to unmonitor
+     * @return boolean representing if removal from monitoring set was successful
+     */
+    public boolean unlogChannel(int channel) {
+        return monitoringChannels.remove(channel);
+    }
+
+    /**
+     * Runs action when channel current draw reaches a threshold. <br>
+     * Note, when channel reaches current draw you will have to re-register the event
+     *
+     * @param channel          to monitor
+     * @param currentThreshold current threshold before runnable is called
+     * @param runnable         action to preform when current is reached
+     * @return if addition to currentEvent list was successful
+     */
+    public boolean onCurrentAboveThreshold(int channel, double currentThreshold, Runnable runnable) {
```

Comment:
`above` and `below` can be replaced with a predicate e.g. `Predicate<current>` to let the caller decide whether they want to trigger or not.